app/src/main/java/cm/aptoide/pt/home/bundles/RemoteBundleDataSource.java

```diff
@@ -170,9 +173,11 @@ public GetStoreWidgetsRequest getMoreBundlesRequest(String url, int offset, int
     body.setOffset(offset);
     return new GetStoreWidgetsRequest(new V7Url(url).remove("getStoreWidgets")
         .get(), body, bodyInterceptor, okHttpClient, converterFactory, tokenInvalidator,
-        sharedPreferences, storeCredentials, clientUniqueId, isGooglePlayServicesAvailable,
-        partnerId, adultContentEnabled, filters, resources, windowManager, connectivityManager,
-        versionCodeProvider, new WSWidgetsUtils(), appBundlesVisibilityManager);
+        sharedPreferences, storeCredentials, idsRepository.getUniqueIdentifier()
+        .toBlocking()
+        .value(), isGooglePlayServicesAvailable, partnerId, adultContentEnabled, filters, resources,
```

Comment:
id could be passed as parameter. The calling method would have to be refactored to something like:

`return idsRepository.getUniqueIdentifier()
        .flatMapObservable(
            id -> getMoreBundlesRequest(newUrl, offset, limit, id).observe(invalidateHttpCache,
                false))`

The same could be potentially done to the adultcontent value.